### DISCUSSION TITLE
ECC Certificate

### DISCUSSION THREAD

**Discussion by microsec** [cypress-io/cypress#30246] [Questions and Help]
Hi! I'm new to cypress but my first step is to connect to a server that authenticated with an ECC certificate. I have my certificate.pem and private_key_ec.pem and even ca.pem in my setup but I got an error that is not RSA and failed to log in. I also checked the documentation https://docs.cypress.io/guides/references/client-certificates but there is nothing about RSA or EC
Cypress mac v13.9.0 

Open SSL: OpenSSL 3.3.0 9 Apr 2024 (Library: OpenSSL 3.3.0 9 Apr 2024)
Failed to load client certificates for clientCertificates[0]: Cannot parse PEM cert: Cannot read public key. OID is not RSA.. For more debug details run Cypress with DEBUG=cypress:server:client-certificates*

Stack trace
Error: Failed to load client certificates for clientCertificates[0]: Cannot parse PEM cert: Cannot read public key. OID is not RSA..  For more debug details run Cypress with DEBUG=cypress:server:client-certificates*
    at Object.S (:4357:21972)
    at Q.open (:4492:9587)
    at C.open (:4492:24943)
    at v.create (:4492:40014)
    at async v.initializeActiveProject (:4357:103960)
    at async Object.onFinalConfigLoaded (:4357:74164)
    at async D.handleSetupTestingTypeReply (:4357:62059)
    at async D.setupNodeEvents (:4357:60660)

My code: "module.exports = {
  clientCertificates: [
    {
      url: '[my website https://...]/**',
      ca: ['certs/ca.pem'],
      certs: [
        {
          cert: 'certs/certificate.pem',
          key: 'certs/private_key_ec.pem',
          passphrase: 'certs/passphrase.txt',
        },
      ],
    }
  ],
  e2e: {
    setupNodeEvents(on, config) {
      // implement node event listeners here
    },
  },
};

it perfectly works with RSA, but my company will stop accepting RSA next year so I would like to move to ECC and I would like to use Cypress, please help.

**Comment by MikeMcC399** [comment]
@microsec 

If you don't get a satisfactory response here, I suggest you report your issue to https://github.com/cypress-io/cypress/issues.
